apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js

```diff
@@ -0,0 +1,299 @@
+    const logEntry = isNull(intentChar) ? message
+        : `${ intentChar.repeat(intentRepeatTimes) } ${ message } ${ intentChar.repeat(intentRepeatTimes) }`
+
+    logger.info({
+        message: logEntry,
+    })
+}
+
+const readMetersPage = async ({ context, offset, pageSize }) => {
+    return await Meter.getAll(
+        context, {}, {
+            sortBy: 'id_ASC',
+            first: pageSize,
+            skip: offset,
+        }
+    )
+}
+
+const readMeterReadings = async ({ context, meters }) => {
+    // first step is get unique meter ids
+    const meterIds = uniq(meters.map(meter => meter.id))
+
+    // then calculate current period window
+    const now = dayjs()
+    const startWindowDate = `${ now.format('YYYY-MM') }-01`
+    const endWindowDate = now.toISOString()
+
+    // load all pages for entity
+    return await loadListByChunks({
+        context,
+        list: MeterReading,
+        where: {
+            date_gte: startWindowDate,
+            date_lte: endWindowDate,
+            meter: {
+                id_in: meterIds,
+            },
+        },
+    })
+}
+
+const readOrganizations = async ({ context, meters }) => {
+    // first step is get unique organizations ids
+    const organizationIds = uniq(meters.map(meter => meter.organization.id))
+
+    // load all pages for entity
+    return await loadListByChunks({
+        context,
+        list: Organization,
+        where: {
+            id_in: organizationIds,
+        },
+    })
+}
+
+const getLocalizedMeterResources = async ({ context, lang }) => {
+    // get context for specific locale
+    const localeDependedContext = {
+        ...context.createContext({ skipAccessControl: true }),
+        req: { locale: lang },
+    }
+
+    // get list of meter resources for certain locale
+    // TODO apply DOMA-3174 mechanic for custom request locale
+    return await loadListByChunks({
+        context,
+        list: MeterResource,
+        where: {},
+    })
+}
+
+const loadAllLocalizedVersionsOfMeterResources = async ({ context }) => {
+    const supportedLanguages = ['en', 'ru']
+    const result = {}
+
+    for (let supportedLanguage of supportedLanguages) {
+        result[supportedLanguage] = await getLocalizedMeterResources({ context, lang: supportedLanguage })
+    }
+
+    return result
+}
+
+const sendSubmitMeterReadingsPushNotifications = async () => {
+    // task implementation algorithm
+    // * Read all meter resources for all supported languages
+    // * Read meters page
+    // * Read meter readings (from 01 day of month to now) for meters above
+    // * Read meter organisation for meters above
+    // * Right join readings/organisation/resources to meters
+    // * Filter meters without readings
+    // * Filter meters without organisation default period
+    // * Join residents to rest of meters
+    // * Send message with specific unique key for set up readings (key is: {meterId}_{residentId}_{now().minusMonth().format(YYYY_MM)})
```

Comment:
probably you need to use meterId_userId ?

Reply:
Not sure if there any difference between those options